pkg/diskmaker/controllers/lvset/create_pv.go

```diff
@@ -0,0 +1,175 @@
+package lvset
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	localv1alpha1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1alpha1"
+	"github.com/openshift/local-storage-operator/pkg/common"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	provCommon "sigs.k8s.io/sig-storage-local-static-provisioner/pkg/common"
+)
+
+func (r *ReconcileLocalVolumeSet) createPV(
```

Comment:
It would be nice to have a version of this function in static-provisioner which can be used directly. To be clear  creating the PV and saving the PV in api-server are two separate concerns and since we preferably want to add labels etc, we just want a unsaved PV to be returned by this function.

Reply:
static-provisioner does have one which we are using inside this function (which refers to create of create/apply, maybe we should go with apply):
```
	localPVConfig := &provCommon.LocalPVConfig{
		Name:            pvName,
		HostPath:        symLinkPath,
		Capacity:        common.RoundDownCapacityPretty(capacityBytes), // d.VolUtil.GetBlockCapacityByte(filePath)
		StorageClass:    storageClass.GetName(),
		ReclaimPolicy:   reclaimPolicy,        // fetch from storageClass (created by LocalVolumeSet operator controller)
		ProvisionerName: r.runtimeConfig.Name, // populate in runtimeconfig earlier
		VolumeMode:      desiredVolumeMode,
		MountOptions:    storageClass.MountOptions, // d.getMountOptionsFromStorageClass(class)
		SetPVOwnerRef:   true,                      // set owner reference from node to PV
		OwnerReference: &metav1.OwnerReference{
			Kind:       r.runtimeConfig.Node.Kind,
			APIVersion: r.runtimeConfig.Node.APIVersion,
			Name:       r.runtimeConfig.Node.GetName(),
			UID:        r.runtimeConfig.Node.GetUID(),
		},
		NodeAffinity: nodeAffinity,
	}
	if desiredVolumeMode == corev1.PersistentVolumeFilesystem && obj.Spec.FSType != "" {
		localPVConfig.FsType = &obj.Spec.FSType
	}
	newPV := provCommon.CreateLocalPVSpec(localPVConfig)
```

Reply:
Yes this is still bit ugly IMO. We can clean this up later. I am thinking we need to make some changes in static-provisioner, so as it can be easy to use as a library. Currently it does not look like the case.